### ISSUE TITLE
Cannot use CONFIG.GO_TOOL in a subrepo

### ISSUE THREAD

**Issue by sagikazarmark** [thought-machine/please#1547]
Looks like subrepos cannot use `CONFIG.GO_TOOL` when it is set to a `go_toolchain`:

```
22:49:22.702   ERROR: ///pleasings2//:go failed:
Parsed build file  but it doesn't contain target go
```

I kind of see why, but at the same time it pretty much limits the usefulness of subrepos (unless I missed something).

**Comment by Tatskaari** [comment]
So you have 

```
[go]
GoTool = //:go|go
```

Where `//:go` is a go toolchain? I guess we add in the subrepo to that when parsing the subrepo packages... not sure if there's an obvious solution to this. Another problem is that we use the main repos config and ignore the .plzconfig of subrepos. I think this might just need a bit of thought.

**Comment by sagikazarmark** [comment]
One more piece of information: in my case it's actually important to use the `GoTool` toolchain because I need the same GOROOT as what the main repo uses.

**Comment by sagikazarmark** [comment]
bump

**Comment by Tatskaari** [comment]
This is going to be solved when we implement #1844 which I believe better captures this issue.
